src/main/java/com/twitter/kinesis/utils/Environment.java

```diff
@@ -145,7 +161,25 @@ private double getDoubleProperty(String propName, double def) {
     return def;
   }
 
+  private Boolean getBooleanProperty(String propName, Boolean def) {
+    String prop = getStringProperty(propName);
+    if (prop != null) {
+      return Boolean.parseBoolean(prop);
+    }
+
+    return def;
+  }
+
   private static String propNameToEnvVar(String propName) {
     return propName.toUpperCase().replace('.', '_');
   }
+
+  private Date getReplayDate(String gnipDateString) {
+    try {
+      return DATE_FORMAT.parse(getStringProperty(gnipDateString));
+    } catch (ParseException e) {
```

Comment:
in both cases (empty and wrong date) we're getting NullPointerException like this:
```java
java.text.ParseException: Unparseable date: "a"
	at java.text.DateFormat.parse(DateFormat.java:357)
	at com.twitter.kinesis.utils.Environment.getReplayDate(Environment.java:179)
	at com.twitter.kinesis.utils.Environment.getReplayFromDate(Environment.java:99)
	at com.twitter.kinesis.ConnectorApplication.endpoint(ConnectorApplication.java:101)
	at com.twitter.kinesis.ConnectorApplication.configure(ConnectorApplication.java:54)
	at com.twitter.kinesis.ConnectorApplication.start(ConnectorApplication.java:87)
	at com.twitter.kinesis.ConnectorApplication.main(ConnectorApplication.java:42)
[main] ERROR com.twitter.kinesis.ConnectorApplication - Unexpected error occured
java.lang.NullPointerException
	at com.google.common.base.Preconditions.checkNotNull(Preconditions.java:213)
	at com.twitter.hbc.core.endpoint.ReplayEnterpriseStreamingEndpoint.<init>(ReplayEnterpriseStreamingEndpoint.java:29)
	at com.twitter.kinesis.ConnectorApplication.endpoint(ConnectorApplication.java:103)
	at com.twitter.kinesis.ConnectorApplication.configure(ConnectorApplication.java:54)
	at com.twitter.kinesis.ConnectorApplication.start(ConnectorApplication.java:87)
	at com.twitter.kinesis.ConnectorApplication.main(ConnectorApplication.java:42)```
and program dies. So I guess it's fine.